question19/js/newsTopics.js

```diff
@@ -0,0 +1,159 @@
+    fragmentTablists.appendChild(tabList);
+  }
+  return fragmentTablists;
+};
+
+//Render Content Element
+const renderContents = (newsData) => {
+  const fragmentContents = document.createDocumentFragment();
+  for (const topics of newsData) {
+    const contentsContainer = createElementWithClassName(
+      "div",
+      "contentsContainer"
+    );
+    const tabContents = createElementWithClassName("div", "tabContents");
+    const tabContentsUl = createElementWithClassName("ul", "tabContentsUl");
+    topics.initialDisplay &&
+      contentsContainer.classList.add("is-active-content");
+
+    fragmentContents
+      .appendChild(contentsContainer)
+      .appendChild(tabContents)
+      .appendChild(tabContentsUl)
+      .appendChild(createTitles(topics));
+    tabContents.appendChild(createImgElements(topics));
+  }
+  contentsWrapper.appendChild(fragmentContents);
+  return contentsWrapper;
+};
+
+//get titles data
+const createTitles = ({ articles }) => {
+  const fragmentTitles = document.createDocumentFragment();
+  for (const topics of articles) {
+    const articleTitle = topics.title;
+    const titleList = document.createElement("li");
+    const articleLink = document.createElement("a");
+    articleLink.href = "#";
+    articleLink.textContent = articleTitle;
+
+    // Display new icon
+    const elapsedTime = getElapsedDays(topics.date);
+    const newArrivalPeriod = 3;
+    if (elapsedTime <= newArrivalPeriod) {
+      const newIcon = createElementWithClassName("span", "new_icon");
+      newIcon.textContent = "new";
+      articleLink.appendChild(newIcon);
+    }
+
+    //add comment icon and number
+    const commentsLength = topics.comments.length;
+    if (commentsLength > 0) {
+      articleLink.appendChild(createCommentIcon(topics));
+    }
+
+    fragmentTitles.appendChild(titleList).appendChild(articleLink);
+  }
+  return fragmentTitles;
+};
+
+//Get img data
+const createImgElements = (newsData) => {
+  const imgWrapper = createElementWithClassName("div", "imgWrapper");
+  const imgTag = document.createElement("img");
+  imgTag.src = newsData.img;
+  imgWrapper.appendChild(imgTag);
+  return imgWrapper;
+};
+
+// Display comment icons and numbers
+const createCommentIcon = (articlesData) => {
+  const commentLength = articlesData.comments.length;
+  const commentIcon = createElementWithClassName("span", "comment_icon");
+  const commentNum = createElementWithClassName("span", "comment_icon_num");
+  const commentIconImg = createElementWithClassName("img", "comment_icon_img");
+  commentNum.textContent = commentLength;
+  commentIconImg.src = "./img/comment_icon.png";
+  commentIcon.appendChild(commentIconImg);
+  commentIcon.appendChild(commentNum);
+  return commentIcon;
+};
+
+// Get the number of days elapsed
+const getElapsedDays = (postDateData) => {
+  const elapsedDays = differenceInDays(new Date(), new Date(postDateData));
+  return elapsedDays;
+};
```

Comment:
IMO
I find it easier to understand when adapted to the method
```suggestion
const getDifferenceDays = (postDateData) => {
  const differenceDays = differenceInDays(new Date(), new Date(postDateData));
  return differenceDays;
};
```

Reply:
Thank you for your review! You are right. I have followed your idea.
fixed commit : https://github.com/csakyo/homework/pull/33/commits/a65ce2c602b7f0de944f112580a0259ea9a0fdac